pages/users/index.tsx

```diff
@@ -2,103 +2,24 @@ import Layout from "../../components/Layout/Layout";
 import styles from "./UsersPage.module.css";
 import FamilyCards from "../../components/Users/FamilyCard/familyCard";
 import { Tabs, Tab } from "@mui/material";
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { getAllUsers } from "../../firebase/firestore/user";
+import { getAllFamilies } from "../../firebase/firestore/family";
 import FullUsersList from "../../components/Users/FullUsersList/fullUsersList";
 
 const UsersPage: React.FunctionComponent = () => {
-  const users = [
-    {
-      address: "2419 Yes Ave",
-      created_at: new Date(),
-      email: "hello@asldkjf.com",
-      family_head: true,
-      family_id: 12345,
-      full_name: "Sydney Bui",
-      last_active: new Date(),
-      parent: true,
-      points: 12000,
-      reward_eligible: true,
-      suspended: false,
-    },
-    {
-      address: "2419 Yes Ave",
-      created_at: new Date(),
-      email: "hello@asldkjf.com",
-      family_head: false,
-      family_id: 12345,
-      full_name: "Bob Builder",
-      last_active: new Date(),
-      parent: false,
-      points: 12000,
-      reward_eligible: true,
-      suspended: false,
-    },
-    {
-      address: "2419 Yes Ave",
-      created_at: new Date(),
-      email: "hello@asldkjf.com",
-      family_head: false,
-      family_id: 12345,
-      full_name: "Hello Hello",
-      last_active: new Date(),
-      parent: false,
-      points: 12000,
-      reward_eligible: true,
-      suspended: false,
-    },
-    {
-      address: "2419 Yes Ave",
-      created_at: new Date(),
-      email: "hello@asldkjf.com",
-      family_head: false,
-      family_id: 12345,
-      full_name: "Bye Hello",
-      last_active: new Date(),
-      parent: false,
-      points: 12000,
-      reward_eligible: true,
-      suspended: false,
-    },
-    {
-      address: "2419 Yes Ave",
-      created_at: new Date(),
-      email: "hello@asldkjf.com",
-      family_head: false,
-      family_id: 12345,
-      full_name: "Oops Hello",
-      last_active: new Date(),
-      parent: false,
-      points: 12000,
-      reward_eligible: true,
-      suspended: false,
-    },
-  ];
-  const families = [
-    {
-      totalPoints: 12000,
-      familyName: "Nguyen",
-      userIds: users,
-      familyId: 12345,
-    },
-    {
-      totalPoints: 12000,
-      familyName: "Nguyen",
-      userIds: users,
-      familyId: 12345,
-    },
-    {
-      totalPoints: 12000,
-      familyName: "Nguyen",
-      userIds: users,
-      familyId: 12345,
-    },
-    {
-      totalPoints: 12000,
-      familyName: "Nguyen",
-      userIds: users,
-      familyId: 12345,
-    },
-  ];
+  const [allUsers, setAllUsers] = useState([]);
+  const [allFamilies, setAllFamilies] = useState([]);
+
+  useEffect(() => {
```

Comment:
Could you look into using getServerSideProps/SSR (https://nextjs.org/docs/basic-features/data-fetching) to load the families for the users page? I have the same thoughts on scalability/response time for this page that I had with Jacob's latest PR. I have an example of this in my auth PR as well.